frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,222 @@
+import {CatalogTile} from 'patternfly-react-extensions/dist/esm/components/CatalogTile';
+import {FilterSidePanel} from 'patternfly-react-extensions/dist/esm/components/FilterSidePanel';
+import {EmptyState} from 'patternfly-react/dist/esm/components/EmptyState';
+import FormControl from 'patternfly-react/dist/esm/components/Form/FormControl';
+
+import {normalizeIconClass} from '../catalog-item-icon';
+
+export class MarketplaceTileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const filters = {
+      byName: {
+        value: '',
+        active: false,
+      },
+      byProvider: {
+        redHat: {
+          label: 'Red Hat',
+          value: 'Red Hat',
+          active: false,
+        },
+        coreOS: {
+          label: 'CoreOS',
+          value: 'CoreOS',
+          active: false,
+        },
+      }
+    };
+
+    let filterCounts = {
+      byProvider: {
+        redHat: 0,
+        coreOS: 0,
+      },
+    };
+
+    filterCounts = this.getFilterCounts(['all'], filters, filterCounts, null);
+
+    this.state = {
+      filters,
+      filterCounts,
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { filters, filterCounts } = this.state;
+    const { items } = this.props;
+
+    if (items !== prevProps.items) {
+      this.setState({items});
+    }
+
+    if (filters !== prevState.filters) {
+      const filteredItems = this.filterItems();
+      this.setState({});
+    }
+
+    // filter counts are updated when filter by name changed
+    if (prevState.filters.byProvider !== filters.byProvider) {
+      this.setState(this.getFilterCounts(['all'], filters, filterCounts));
+    }
+  }
+
+  getFilterCounts(activeTabs, filters, filterCounts, allCategories=null) {
+    const { items } = this.props;
+    const count = _.countBy(items, 'tileProvider');
+
+    const newFilterCounts = {...filterCounts};
+    for (const provider in newFilterCounts.byProvider) {
+      newFilterCounts.byProvider[provider] = count[provider] || 0;
+    }
+    return newFilterCounts;
+  }
+
+  filterItems() {
+    const { filters } = this.state;
+    const { byName, byProvider } = filters;
+    const { items } = this.props;
+
+    if (!byProvider.redHat.active && !byProvider.coreOS.active && !byName.active) {
+      return items;
+    }
+
+    let filteredItems = [];
+
+    if (byProvider.redHat.active) {
+      filteredItems = _.filter(items, (item) => item.tileProvider.includes(byProvider.redHat.value));
+    }
+
+    if (byProvider.coreOS.active) {
+      filteredItems = filteredItems.concat(_.filter(items, (item) => item.tileProvider.includes(byProvider.coreOS.value)));
+    }
+
+    if (byName.active) {
+      const filterString = byName.value.toLowerCase();
+      return _.filter((byProvider.redHat.active || byProvider.coreOS.active ? filteredItems : items), item => item.tileName.toLowerCase().includes(filterString));
+    }
+
+    return filteredItems;
+  }
```

Comment:
Not sure if this is stated anywhere but when you filter, is this a subsection or a selection? i.e. when you click both red hat and coreos, should you get back all operators provided by coreos AND red hat (i.e. intersection) or coreos OR red hat (i.e. union)?

Reply:
For the checkboxes it's "union" rather than "intersection". I think this makes sense since operators can only match to one provider each. However, name filtering only compares against the tiles listed given the already applied filters.

Reply:
For one category, the filters act as union. However, for multiple categories I'm leaning towards intersection.
My implementation here filters the items category by category, and uses `_.union` for checkboxes in the same category to get rid of any duplicates. I have a gut feeling there's a more optimal implementation, since I end up using `_.sort` on the items in the return.

Reply:
Yeah this complicates the implementation if we have intersection across categories but union within a category...

From the mocks @tlwu2013 provided, there are numbers next to the checkbox labels indicating the number of items with that label. I think we can assume for now that checking a given box should return **at least** the number indicated on the side. This means either we re-render the numbers during every filter (which is not very efficient) or we just stick with union. What do you think?

Reply:
> I think we can assume for now that checking a given box should return **at least** the number indicated on the side. This means either we re-render the numbers during every filter

Hi, for Catalog, the 'byType' filter counts get updated whenever a new Category/subcategory is selected or whenever the Name filter changes, so clicking on a filter checkbox should return the exact number shown in the filter counts.  Ex. A drill down into 'Languages -> Javascript' may show byType filter counts of '[ ] Service Class (5)' and '[ ] Images (4)'.   If I then type 'N' into the Name filter textbox, counts may update to '[ ] Service Class (2)' and '[ ] Images (1)'.  At this point checking '[X] Service Class (2)' would show exactly two tiles matching: 'category === 'Javascript' && title contains 'N' & type === service_class'.   
I would image another set of checkbox's (byProvider) filter counts would have to updated to show counts for all current active filters, in this case byProviders counts updated when a byType is checked/unchecked (and visa-versa; byType's counts update when byProvider checked).
At least that's the way I see it, it's all about drilling down to specific criteria, and counts get updated along the way, showing what's available, at the time, if checked.

Reply:
Gotcha! That makes a lot of sense - thanks for the clarification

Reply:
@dtaylor113 So what you're saying is that filterCounts should be updated whenever the filters change? For example, with just byProvider, if I check "Red Hat" then the counts would be "Red Hat (3)" and "CoreOS (0)" and checking "CoreOS" would change the counts to  "Red Hat (3)" and "CoreOS (2)"? Just want to make sure I'm following.

Reply:
This seems to be pretty close to the filtering counts functionality we're looking for: [BestBuy TVs](https://www.bestbuy.com/site/tvs/60-inch-tvs/pcmcat1514910288859.c?id=pcmcat1514910288859)